orderbook/usecase/orderbook_usecase_test.go

```diff
@@ -0,0 +1,314 @@
+package orderbookusecase_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	cltypes "github.com/osmosis-labs/osmosis/v25/x/concentrated-liquidity/types"
+	"github.com/osmosis-labs/sqs/domain/mocks"
+	orderbookdomain "github.com/osmosis-labs/sqs/domain/orderbook"
+	orderbookusecase "github.com/osmosis-labs/sqs/orderbook/usecase"
+	"github.com/osmosis-labs/sqs/sqsdomain/cosmwasmpool"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/osmosis/v25/app/apptesting"
+)
+
+type OrderbookUsecaseTestSuite struct {
+	apptesting.ConcentratedKeeperTestHelper
+}
+
+func TestOrderbookUsecaseTestSuite(t *testing.T) {
+	suite.Run(t, new(OrderbookUsecaseTestSuite))
+}
+
+func (s *OrderbookUsecaseTestSuite) TestCreateFormattedLimitOrder() {
+	testCases := []struct {
+		name          string
+		order         orderbookdomain.Order
+		quoteAsset    orderbookdomain.Asset
+		baseAsset     orderbookdomain.Asset
+		setupMocks    func(orderbookRepo *mocks.OrderbookRepositoryMock, tokensUsecase *mocks.TokensUsecaseMock)
+		expectedError string
+		expectedOrder orderbookdomain.LimitOrder
+	}{
+		{
+			name: "tick not found",
+			order: orderbookdomain.Order{
+				TickId: 99, // Non-existent tick ID
+			},
+			expectedError: "tick not found",
```

Comment:
That is a valid point, my initial attempt implementing that failed, since for this method client(s) specific errors are irrelevant, and added complexity by defining errors and introduced additional code outweights the benefts, thus I would leave as it is at least for now. Let me know if you still think we should avoid doing that.

Reply:
I still think it is worth it doing something like [this](https://github.com/osmosis-labs/sqs/blob/f1740c12a07ac7a2c9df284d45b1323224a755ef/domain/errors.go#L318) so that if an error message is ever updated, we only need to update it once and don't worry about several hardcoded tests

Reply:
Thanks, I have implemented as per your suggestion [here](https://github.com/osmosis-labs/sqs/pull/492/files#diff-a585202576c1e374948b8b760a1fe96534e5a35a29ccb3e846e5a7548c714f0a). Would that work? :pray:

Reply:
yes, nice job!